### PR TITLE
Fix a bug in the syntax generator with VB constructors and attributes.

### PR DESCRIPTION
**Pull request by 333fred**
Fixes https://github.com/dotnet/roslyn/issues/31720. @dotnet/roslyn-compiler for a very quick review.

### CODE REVIEW

**Comment by 333fred** [comment]
FYI @sharwell.

**Comment by CyrusNajmabadi** [comment]
No need for compiler review.  This is entirely on the IDE side.

File: src/Workspaces/VisualBasic/Portable/CodeGeneration/VisualBasicSyntaxGenerator.vb [hunk 1]
```diff
@@ -1769,12 +1769,10 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.CodeGeneration
                 Case SyntaxKind.FunctionBlock,
                      SyntaxKind.SubBlock,
                      SyntaxKind.ConstructorBlock
-                    Return DirectCast(node, MethodBlockSyntax).BlockStatement.AttributeLists
+                    Return DirectCast(node, MethodBlockBaseSyntax).BlockStatement.AttributeLists
                 Case SyntaxKind.FunctionStatement,
                      SyntaxKind.SubStatement
                     Return DirectCast(node, MethodStatementSyntax).AttributeLists
-                Case SyntaxKind.ConstructorBlock
-                    Return DirectCast(node, ConstructorBlockSyntax).BlockStatement.AttributeLists
```
Comment (CyrusNajmabadi) [inline src/Workspaces/VisualBasic/Portable/CodeGeneration/VisualBasicSyntaxGenerator.vb]: alternative approach would be to keep this and just remove SyntaxKind.ConstructorBlock from above.  But i'm ok with either.

Comment (CyrusNajmabadi) [reply src/Workspaces/VisualBasic/Portable/CodeGeneration/VisualBasicSyntaxGenerator.vb]: if you do go this path, i'd prefer all the MethodBlockBase guys go here (like operators and accessors).  but i don't feel strongly.

**Comment by mavasani** [comment]
Tagging @jinujoseph for approval

**Pull request merged**